docs/concepts/models.md

```diff
@@ -168,6 +171,53 @@ Models possess the following methods and attributes:
     See [Changes to `pydantic.BaseModel`](../migration.md#changes-to-pydanticbasemodel) in the
     [Migration Guide](../migration.md) for details on changes from Pydantic V1.
 
+## Data conversion
+
+Pydantic may cast input data to force it to conform to model field types,
+and in some cases this may result in a loss of information.
+For example:
+
+```python
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    a: int
+    b: float
+    c: str
+
+
+print(Model(a=3.000, b='2.72', c=b'binary data').model_dump())
+#> {'a': 3, 'b': 2.72, 'c': 'binary data'}
+```
+
+This is a deliberate decision of Pydantic, and is frequently the most useful approach. See
+[here](https://github.com/pydantic/pydantic/issues/578) for a longer discussion on the subject.
+
+This is also the case for collections. In most cases, you shouldn't make use of abstract container classes
+and just use a concrete type, such as [`list`][]:
+
+```python
+from typing import List
+
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    items: List[int]  # (1)!
+
+
+print(Model(items=(1, 2, 3)))
+#> items=[1, 2, 3]
+```
+
+1. In this case, you might be tempted to use the abstract [`Sequence`][collections.abc.Sequence] type
+   to allow both lists and tuples. But Pydantic takes care of converting the tuple input to a list, so
+   in most cases this isn't necessary.
+
+Nevertheless, Pydantic provides a [strict mode](strict_mode.md), where no data conversion is performed.
+Values must be of the same type than the declared field type.
```

Comment:
Let's move this up above your new example closer to where it was initially added